utils/yoke/src/yoke.rs

```diff
@@ -271,6 +271,27 @@ impl<Y: for<'a> Yokeable<'a>, C> Yoke<Y, C> {
         self.cart
     }
 
+    /// Unsafe function for replacing the cart with another
+    ///
+    /// This can be used for type-erasing the cart, for example.
+    ///
+    /// # Safety
+    ///
+    /// - `f()` must not panic
+    /// - References from the yokeable `Y` should still be valid for the lifetime of the
+    ///   returned cart type.
+    ///
+    /// Typically, this means implementing `f` as something which _wraps_ the inner cart type.
+    /// `Yoke` only really cares about destructors for its carts so it's fine to erase other
+    /// information about the cart, as long as the backing data will still be destroyed at the
+    /// same time.
+    pub unsafe fn replace_cart<C2>(self, f: impl FnOnce(C) -> C2) -> Yoke<Y, C2> {
```

Comment:
Suggestion: Make this function less general; make it only do the trait object casting.

Reply:
This ties into my comment above about moving the trait object into this crate